### PR TITLE
fix http_options in requests

### PR DESCRIPTION
**Pull request by zot**
Was getting an error trying to set http_options; this change makes them work

### CODE REVIEW

**Review by mattBrzezinski — APPROVED** [review]
bors try

**Comment by zot** [comment]
Checking the tests...


-- Bill

**Comment by zot** [comment]
bors try

**Comment by zot** [comment]
Guess I can't do that :)

**Comment by mattBrzezinski** [comment]
bors try

File: src/utilities/request.jl [hunk 1]
```diff
@@ -10,7 +10,7 @@ Base.@kwdef mutable struct Request
 
     return_stream::Bool=false
     response_stream::Union{<:IO, Nothing}=nothing
-    http_options::Array{String}=[]
+    http_options::Dict{Symbol,<:Any} = Dict{Symbol,String}()
```
Comment (mattBrzezinski) [inline src/utilities/request.jl:13]:
```suggestion
    http_options::AbstractDict{Symbol,<:Any}=LittleDict{Symbol,String}()
```

File: src/utilities/utilities.jl [hunk 2]
```diff
@@ -65,7 +65,7 @@ function _extract_common_kw_args(service, args)
         return_raw=_pop!(args, "return_raw", false),
         response_stream=_pop!(args, "response_stream", nothing),
         headers=LittleDict{String, String}(_pop!(args, "headers", [])),
-        http_options=_pop!(args, "http_options", []),
+        http_options=_pop!(args, "http_options", Dict{Symbol, String}()),
```
Comment (mattBrzezinski) [inline src/utilities/utilities.jl:68]:
```suggestion
        http_options=_pop!(args, "http_options", LittleDict{Symbol, String}()),
```

**Review by mattBrzezinski — APPROVED** [review]
Just for consistencies as other places use LD everywhere.

**Comment by kogbill** [comment]
I tried that but using LD produces errors during the conversion calls.

**Comment by zot** [comment]
Sorry, "kogbill" up there is me.

I tried using LittleDict but it causes conversion errors with http_options.

**Comment by mattBrzezinski** [comment]
> Sorry, "kogbill" up there is me.
> 
> I tried using LittleDict but it causes conversion errors with http_options.

I just pulled down the branch, implemented the suggested changes above and it work. I think you may have not loosened the restriction on the `Request.http_options` type from `Dict -> AbstractDict` which would have cause an error when trying to default to a `LittleDict`.

**Comment by kogbill** [comment]
OK, whatever changes you want to make are just fine with me!

**Review by mattBrzezinski — APPROVED** [review]
bors r+

**Pull request merged**